draft-schinazi-quic-version-negotiation.md

```diff
@@ -112,42 +115,143 @@ versions as long as they are reasonably popular.  This is not generally an issue
-            QuicVersion negotiated_compatible_version;
-      }
-} CompatibleVersions;
-~~~~
-
-The client's "supported_compatible_versions" parameter lists the versions it
-supports in decreasing order of preference. The server's
-"negotiated_compatible_version" parameter lists the version it has selected.
-If the client does not send this transport parameter, the server MUST assume
-that the client only supports the version it used for the
-Initial packet and MUST NOT send its own parameter.
-
-Clients MAY include versions following the pattern 0x?a?a?a?a in their
-supported_compatible_versions. Those versions are reserved to exercise
-version negotiation (see the Versions section of {{I-D.ietf-quic-transport}}),
-and MUST be ignored by the server when parsing supported_compatible_versions.
+# Version Negotiation Transport Parameter
+
+This document registers a new transport parameter, `version_negotiation`. The
+contents of this transport parameter depend on whether the client or server is
+sending it, and are shown below:
+
+~~~
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                    Attempted Version (32)                     |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|            Received Negotiation Version Count (i)           ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|              Received Negotiation Version 1 (32)              |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|             [Received Negotiation Version 2 (32)]             |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+                               ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|             [Received Negotiation Version N (32)]             |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                 Compatible Version Count (i)                ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                   Compatible Version 1 (32)                   |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                  [Compatible Version 2 (32)]                  |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+                               ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                  [Compatible Version N (32)]                  |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+~~~
+{: #fig-client-tp title="Client Transport Parameter Format"}
+
+The content of each field is described below:
+
+Attempted Version:
+
+: The version that the client is using in this Initial. This field MUST be
+  equal to the value of the Version field in the long header that carries
+  this transport parameter.
```

Comment:
Why is this needed?

Reply:
It prevents a class of attacks where the CRYPTO frame is copied from one version to another, by adding it to the TLS transcript. See Security Considerations.

Reply:
Did you forget to write that text? I dont' see it.

Reply:
Good point, added normative validation text